src/mapeo-manager.js

```diff
@@ -0,0 +1,194 @@
+import { KeyManager } from '@mapeo/crypto'
+import Database from 'better-sqlite3'
+import { eq } from 'drizzle-orm'
+import { drizzle } from 'drizzle-orm/better-sqlite3'
+import { migrate } from 'drizzle-orm/better-sqlite3/migrator'
+import { IndexWriter } from './index-writer/index.js'
+import { MapeoProject } from './mapeo-project.js'
+import { projectKeysTable, projectTable } from './schema/client.js'
+import { NAMESPACES } from './core-manager/index.js'
+
+/** @typedef {import('./types.js').ProjectKeys} ProjectKeys */
+
+const CLIENT_SQLITE_FILE_NAME = 'client.db'
+
+export class MapeoManager {
+  #keyManager
+  #projectSettingsIndexWriter
+  #storagePath
+  #db
+
+  /**
+   * @param {Object} [opts]
+   * @param {string} [opts.storagePath] Folder for all data storage (hypercores and sqlite db). Folder must exist. If not defined, everything is stored in-memory
+   */
+  constructor({ storagePath } = {}) {
+    const dbPath =
+      storagePath !== undefined
+        ? path.join(storagePath, CLIENT_SQLITE_FILE_NAME)
+        : ':memory:'
+
+    const sqlite = new Database(dbPath)
+    this.#db = drizzle(sqlite)
+    migrate(this.#db, { migrationsFolder: './drizzle/client' })
+
+    this.#keyManager = new KeyManager(KeyManager.generateRootKey())
+    this.#projectSettingsIndexWriter = new IndexWriter({
+      tables: [projectTable],
+      sqlite,
+    })
+    this.#storagePath = storagePath
+  }
+
+  /**
+   * Create a new project.
+   * @param {import('type-fest').Simplify<Partial<Omit<import("@mapeo/schema").ProjectValue, 'schemaName'>>>} settings
+   * @returns {Promise<string>}
+   */
+  async createProject(settings) {
+    // 1. Create project keypair
+    const projectKeypair = {
+      publicKey: randomBytes(32),
+      secretKey: randomBytes(64),
+    }
+
+    // 2. Create namespace encryption keys
+    /** @type {Record<import('./core-manager/core-index.js').Namespace, Buffer>} */
+    const namespaceEncryptionKeys = {
+      auth: randomBytes(32),
+      blob: randomBytes(32),
+      blobIndex: randomBytes(32),
+      config: randomBytes(32),
+      data: randomBytes(32),
+    }
+
+    // 3. Save keys to client db in projectKeys table
+    /** @type {ProjectKeys} */
+    const keys = {
+      projectKey: projectKeypair.publicKey,
+      projectSecretKey: projectKeypair.secretKey,
+      authEncryptionKey: namespaceEncryptionKeys.auth,
+      blobEncryptionKey: namespaceEncryptionKeys.blob,
+      blobIndexEncryptionKey: namespaceEncryptionKeys.blobIndex,
+      configEncryptionKey: namespaceEncryptionKeys.config,
+      dataEncryptionKey: namespaceEncryptionKeys.data,
+    }
+
+    // TODO: Update to use @mapeo/crypto when ready (https://github.com/digidem/mapeo-core-next/issues/171)
+    const projectId = projectKeypair.publicKey.toString('hex')
+
+    this.#db
+      .insert(projectKeysTable)
+      .values({
+        projectId,
+        encryptionKeys: this.#keyManager.encryptLocalMessage(
```

Comment:
Naming suggestion. These aren't actually all encryption keys
```suggestion
        keysCipher: this.#keyManager.encryptLocalMessage(
```

Reply:
addressed via 6d14bacb6f178af5102c8d855efffbb0d3b4c085